pkg/context/httpcontext.go

```diff
@@ -206,10 +249,40 @@ func (ctx *httpContext) Span() tracing.Span {
 	return ctx.span
 }
 
+// Fill the empty tag. If all existing Tags are filled, create new one.
 func (ctx *httpContext) AddTag(tag string) {
 	ctx.tags = append(ctx.tags, tag)
 }
 
+// Add new LazyTag.
+func (ctx *httpContext) AddLazyTag(ns string, prefix string, msg string, intMsg int) {
+	lazyTag := lazyTagPool.Get().(*LazyTag)
+	lazyTag.Namespace = ns
+	lazyTag.Prefix = prefix
+	lazyTag.StringMsg = msg
+	lazyTag.IntMsg = intMsg
+	lazyTag.Sep = ": "
+	ctx.lazyTags = append(ctx.lazyTags, lazyTag)
+}
+
+func (ctx *httpContext) ReleaseLazyTags() {
+	for _, lazyTag := range ctx.lazyTags {
+		lazyTagPool.Put(lazyTag)
+	}
+}
+
+// Return all tags in string format.
+func (ctx *httpContext) GetTags() []string {
+	allTags := make([]string, len(ctx.tags)+len(ctx.lazyTags))
+	for i := 0; i < len(ctx.tags); i++ {
+		allTags[i] = ctx.tags[i]
+	}
+	for i := 0; i < len(ctx.lazyTags); i++ {
+		allTags[len(ctx.tags)+i] = ctx.lazyTags[i].String()
+	}
+	return allTags
+}
```

Comment:
Please make sure no one cares about the order of tags, compare to the logic before, the order of the tags is different.

@xxx7xxxx could you please help confirm this?

Reply:
the tags is used for debug... I think the order of tags is important, use tags we can see the filter exec order of pipeline. @samutamm Is that a good idea we use same struct for tags and lazy tags, so we can keep the order of tags is the order of filter exec?

Reply:
Now the order of the tags should be the same as before. There is only on slice for tags that is appended in order.